execution/function/functions.go

```diff
@@ -46,6 +46,16 @@ func simpleFunc(f func(float64) float64) FunctionCall {
 
 }
 
+func extractFloatOnlySamples(samples []promql.Sample) []promql.Sample {
+	var res []promql.Sample
+	for _, sample := range samples {
+		if sample.H == nil {
+			res = append(res, sample)
+		}
+	}
+	return res
```

Comment:
I am actually thinking, can we do this in place so that we don't have to alloc and grow the slice? https://zetcode.com/golang/filter-slice/

But not a blocker, just an optimization.

Reply:
Why do we need to call this function for `idelta` and `irate` now? Was something changed in the Prometheus engine?

Reply:
Because of this change https://github.com/prometheus/prometheus/blob/main/promql/functions.go#L250. In previous Prometheus both float and histogram are inside slice of points. If all points are histogram value, it will not return empty vector instead it will return a vector with all histogram value set to nil. However, in new Prometheus version, float and histogram points are in separate slices, this function checks the size of float points. In the case that only histogram points exist, it will return empty vector which is different from the old version. That is why I need to add the filtering logic here to make sure empty vector got returned if only histogram points exist.